tests/test.py

```diff
@@ -0,0 +1,106 @@
+import unittest
+from selenium import webdriver
+from selenium.webdriver.common.keys import Keys
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.ui import Select
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.common.exceptions import NoSuchElementException
+from selenium.webdriver.support import expected_conditions as EC
+import subprocess
+import random
+
+class CodalabTestCase(unittest.TestCase):
+
+    def run_command(self, args, expected_exit_code=0):
+        try:
+            output = subprocess.check_output(args)
+            exitcode = 0
+        except subprocess.CalledProcessError, e:
+            output = e.output
+            exitcode = e.returncode
+        print '>> %s (exit code %s, expected %s)\n%s' % (args, exitcode, expected_exit_code, output)
+        assert expected_exit_code == exitcode, 'Exit codes don\'t match'
+        return output.rstrip()
+
+    def setUp(self):
+        self.browser = webdriver.Firefox()
+        self.addCleanup(self.browser.quit)
+        self.base_url = "http://localhost:8000/"
+
+    def testBaseURL(self):
+        self.browser.get(self.base_url)
+        self.browser.maximize_window()
+
+        # Make sure that there is no error displayed on the page
+        self.assertNotIn('ERROR', self.browser.page_source)
+        # check title
+        self.assertIn('CodaLab - Home', self.browser.title)
+
+        for item in ['Worksheets', 'Competitions', 'Help', 'Example Worksheets', 'Latest Competitions']:
+            self.assertIn(item, self.browser.page_source)
+
+    def testWorksheetContent(self):
+        cl = 'cl'
+        self.browser.get(self.base_url)
+        self.browser.maximize_window()
+        
+        # create a temp worksheet for testing
+        temp_worksheet_name = 'test-codalab-' + str(random.randint(0, 1000000))
+        temp_worksheet_title =  'The really long testing title for testing the automated testing of worksheets'
+        temp_worksheet_uuid = self.run_command([cl, 'new', temp_worksheet_name])
+        temp_file_name = 'cl-web-interface-testing.txt'
+        temp_worksheet_paragraph_content = "For running automated testing on the cli " * 10
+        self.run_command([cl, 'work', temp_worksheet_uuid])
+        self.run_command([cl, 'wedit', '-t', temp_worksheet_title])
+        self.run_command([cl, 'add',  '-m', temp_worksheet_paragraph_content])
+        self.run_command(['touch', temp_file_name])
+        temp_file_uuid = self.run_command([cl, 'upload', 'dataset', temp_file_name])
+
+        # click the worksheet button on the top right of the main page
+        self.browser.find_element_by_xpath("html/body/nav/div/div[2]/ul/li[1]/a").click()
```

Comment:
yup. But that would break if there are multiple instances of word 'Worksheets'. That is why I was doing it this way. I will make the changes.